com.unity.multiplayer.mlapi/Runtime/Messaging/RPCQueue/RPCQueueContainer.cs

```diff
@@ -481,10 +477,18 @@ public void ClearLoopBackWriter(QueueHistoryFrame.QueueFrameType queueFrameType,
 
             //Write a filler dummy size of 0 to hold this position in order to write to it once the RPC is done writing.
             queueHistoryItem.queueWriter.WriteInt64(0);
+
             if (NetworkingManager.Singleton.IsHost && queueFrameType == QueueHistoryFrame.QueueFrameType.Inbound)
             {
-                queueHistoryItem.queueWriter.WriteInt64(1); //Write the stream position offset for inbound as 1
-                queueHistoryItem.hasLoopbackData = true; //The only case for this is when it is the Host
+                if (!IsUsingBatching())
+                {
+                    queueHistoryItem.queueWriter.WriteInt64(1);
```

Comment:
Let's put a simple comment as to what we are writing here. As a general note, I am not a fan of magic numbers, prefer descriptive constants.
Also, we could shorten this sort of thing up a bit:
  var descriptiveTagForThisValue = (IsUsingBatching())? 1: 0:
  queueHistoryItem.queueWriter.WriteInt64(descriptiveTagForThisValue);